src/utils/__tests__/documentdb.test.ts

```diff
@@ -739,4 +739,58 @@ describe("fromQueryEither", () => {
     expect(result.isLeft()).toBeTruthy();
     expect(result.value).toEqual({ code: "error", body: errorMessage });
   });
+
+  it("should return the correct query error when a QueryError is thrown", async () => {
+    interface IExpectedResultType {
+      foo: string;
+    }
+
+    const cosmosdbError: DocumentDb.QueryError = {
+      body: "Bad Request",
+      code: 400
+    };
+    const queryFn: () => Promise<
+      Either<DocumentDb.QueryError, IExpectedResultType>
+    > = async () => {
+      throw cosmosdbError;
```

Comment:
it's not clear, AFAIK the cosmosdb error is never throw but returned in the left side of the either